### PR TITLE
Fix issue where ens resolution stops working after sw is terminated

### LINKED ISSUE #16323: [Bug]: [MV3] ENS resolution stopped working after terminating service worker
### Describe the bug

**Problem**: when I terminate the service worker, after it re-starts, ENS resolution stopped working and when I introduce an ENS domain, no address is ever resolved anymore.

https://user-images.githubusercontent.com/54408225/199038834-7768a0ef-658d-4cef-aee3-cd43d09f6fce.mp4

### Steps to reproduce

1. Unlock MM
2. Go to Settings // Contacts
3. Click Add Contact
4. Enter a valid ENS, to make sure it exists *i.e. vitalik.eth
5. Delete the input you just filled
6. Stop the service worker
7. Introduce again the valid ENS name -- see how ENS domain is never resolved

### Error messages or log output

_No response_

### Version

Tested on top of this PR https://github.com/MetaMask/metamask-extension/pull/15558

### Build type

_No response_

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adonesky1**
## Explanation
When the service worker terminates after an ENS slice has been initialized, the `provider` proxy object that [is instantiated as part of the slice initialization](https://github.com/MetaMask/metamask-extension/blob/develop/ui/ducks/ens.js#L145) becomes stale and unresponsive. This causes an issue where ENS names are not resolved when entered in an ENS input if the service worker dies while a user is on a page where the ENS input lives, and enters the ENS name in the input after the service worker terminates: 

https://user-images.githubusercontent.com/34557516/199115850-67793f3d-e35a-426e-ac97-d84d21cd7f57.mp4

~this PR resolves this issue by timing out the request to the provider to resolve the ENS name if it's hanging, throwing an error upon timeout, catching that error, re-initializing the ENS slice (and with it the provider) and then attempting again to resolve the ENS name.~

**Found a simpler approach:**
The `writable` property on the `provider` object on the `web3Provider` flips to false when stale/unresponsive, so we don't need to go through the process of attempting to resolve a name and seeing if the attempt times out and then re-initializing. Now we can just check if the `writable` property is false and if it is reinitialize the slice and the provider.

Behavior with the fix:

https://user-images.githubusercontent.com/34557516/199116361-48b5ee5b-da20-4261-9295-0409c208aef9.mov

## More Information

* Fixes #16323

## Manual Testing Steps

1. Unlock MM
2. Go to Settings // Contacts
3. Click Add Contact
4. Enter a valid ENS, to make sure it exists *i.e. vitalik.eth
5. Delete the input you just filled
6. Stop the service worker
7. Enter a valid ENS name again -- see that the ENS address is still resolved

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/ducks/ens.js [hunk 1]
```diff
@@ -177,9 +187,25 @@ export function lookupEnsName(ensName) {
       let address;
       let error;
       try {
```
Comment (adonesky1) [inline ui/ducks/ens.js:189]: seems like another viable approach here is 
```
const isProviderActive = provider.provider.writable;
        if (!isProviderActive) {
          await dispatch(initializeEnsSlice());
        }
        address = await provider.resolveName(trimmedEnsName);
```

the `writable` property on the `provider` object on the `web3provider` will flip to false when stale/unresponsive, so we don't need to go through the process of attempting to resolve a name and seeing if the attempt times out and then re-initializing. This approach is more streamlined.

Comment (adonesky1) [reply ui/ducks/ens.js:180]: I've adopted this approach [here](https://github.com/MetaMask/metamask-extension/pull/16335/commits/3f95f2e5c72db98f57d7521a598428f83d055cee)

**Review by jpuri — APPROVED** [review]

File: ui/ducks/ens.js [hunk 2]
```diff
@@ -177,7 +180,13 @@ export function lookupEnsName(ensName) {
...
-        address = await provider.resolveName(trimmedEnsName);
+        // the writable property on the 'provider' object on the 'web3Provider' flips to false when stale
```
Comment (darkwing) [inline ui/ducks/ens.js:183]: Biggest nit of all time but if we're going to capitalize `This` on the next line, maybe we capitalize `The` here?

**Review by darkwing — APPROVED** [review]
Tested and works great.  Nice job @adonesky1 !

**Pull request merged**